pkg/scheduler/internal/cache/cache.go

```diff
@@ -150,6 +150,19 @@ func (cache *schedulerCache) List(selector labels.Selector) ([]*v1.Pod, error) {
 	return cache.FilteredList(alwaysTrue, selector)
 }
 
+func (cache *schedulerCache) Search(selectors []labels.Selector) ([]*v1.Pod, error) {
+	matchAllSelctors := func(p *v1.Pod) bool {
+		for _, selector := range selectors {
+			if !selector.Matches(labels.Set(p.Labels)) {
+				return false
+			}
+		}
+		return true
+	}
+
+	return cache.FilteredList(matchAllSelctors, selectors[0])
```

Comment:
the length of selectors is always bigger than 0?

Reply:
Yes it is checked in CalculateSpreadPriorityReduce which calls Search as far as I read.

Reply:
Thanks for the review. Added a `len(selectors) == 0` check here too.